### ISSUE TITLE
Upgrade does not remove old validatingwebhookconfiguration

### ISSUE THREAD

**Issue by dvob** [istio/istio#35244]
### Bug Description

If we upgrade from 1.10.2 to 1.11.2 a new validating webhook configuration is created but the old one is not removed.
Steps to reproduce:
```bash
kind create cluster
istioctl_1.10.2 install
kubectl get validatingwebhookconfigurations # shows istiod-istio-system
istioctl_1.11.2 upgrade
kubectl get validatingwebhookconfigurations # shows istiod-istio-system istio-validator-istio-system
```

If we directly install 1.11 there is only `istio-validator-istio-system`:
```bash
kind create cluster
istioctl_1.11.2 install
kubectl get validatingwebhookconfigurations # only shows istio-validator-istio-system
```

### Version

```prose
istioctl versions: 1.10.2 and 1.11.2

$ kubectl version --short
Client Version: v1.21.0
Server Version: v1.21.1
```

### Affected product area

- [ ] Docs
- [X] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [X] Upgrade
- [ ] Multi Cluster
- [ ] Virtual Machine
- [ ] Control Plane Revisions

### Is this the right place to submit this?

- [X] This is not a security vulnerability
- [X] This is not a question about how to use Istio

**Comment by soumyaparui** [comment]
Similar issue faced even while doing canary upgrade using version.
`kubectl get validatingwebhookconfiguration
NAME                                             WEBHOOKS   AGE
istio-validator-1-11-2-distroless-istio-system   2          3h24m
istiod-istio-system                              1          21h
`

**Comment by yckbilly1929** [comment]
similar case when i do canary upgrade from 1.11.0 to 1.11.3 by using revision labels approach

the old validator webhook still exists after old control plane deleted, like this:

```
NAME                                                              WEBHOOKS   AGE
flowcontrol-guardrails.config.common-webhooks.networking.gke.io   1          104d
istio-validator-1-11-0-istio-system                               2          46d
istio-validator-1-11-3-istio-system                               2          12h
```

**Comment by mapkon** [comment]
This is a similar case when we downgrade to 1.10.4 from 1.11.3. The new validatingwebhookconfiguration `istio-validator-1-11-3-istio-system` still exists after the 1.11.3 control plan is deleted and downgraded to `1.10.4-xxx`

**Comment by Dylan-Christopherson** [comment]
In discussions with Tong, I have some follow up questions to add. 

We are using the in cluster operator deployment based upgrade instead of the istioctl install/upgrade based upgrade. This problem exists using this method. 

What do these webhooks do (`istiod-istio-system` and `istio-validator-istio-system`?
What is the difference between them?
Why is the `istiod-istio-system` webhook remaining?
What happens if the `istiod-istio-system` webhook is deleted in 1.10 prior to installing 1.11?
Does a safe upgrade require waiting until 1.11 is installed before deleting the istiod-istio-system  webhook?
Is it safe to delete the istiod-istio-system before an install of 1.12?

**Comment by yckbilly1929** [comment]
> similar case when i do canary upgrade from 1.11.0 to 1.11.3 by using revision labels approach
> 
> the old validator webhook still exists after old control plane deleted, like this:
> 
> ```
> NAME                                                              WEBHOOKS   AGE
> flowcontrol-guardrails.config.common-webhooks.networking.gke.io   1          104d
> istio-validator-1-11-0-istio-system                               2          46d
> istio-validator-1-11-3-istio-system                               2          12h
> ```

This issue seems to be solved in 1.12. When I upgrade from 1.11.3 to 1.12 this time, no more zombie webhooks.

Not sure if it relates to a different operation step this time that I rollout-restarted deployments and statefulsets before deleting the previous version.

**Comment by dvob** [comment]
> This issue seems to be solved in 1.12. When I upgrade from 1.11.3 to 1.12 this time, no more zombie webhooks.

The problem is not solved with 1.12.x.

If you directly install 1.11.3 and then upgrade to 1.12.0 you don't have the problem because the old validating webhook which does not get removed is from <=1.10.x.
